corehq/apps/app_manager/models.py

```diff
@@ -3845,7 +3848,88 @@ def validate_for_build(self):
         return errors
 
 
-class VersionedDoc(LazyAttachmentDoc):
+class LazyBlobDoc(BlobMixin, LazyAttachmentDoc):
+    """Adapt LazyAttachmentDoc to blob db"""
+
+    migrating_blobs_from_couch = True
+    __save_params = None
+
+    @classmethod
+    def wrap(cls, data):
+        if "_attachments" in data:
+            data = data.copy()
+            attachments = data.pop("_attachments")
+        else:
+            attachments = None
+        self = super(LazyAttachmentDoc, cls).wrap(data)
+        if attachments:
+            for name, attachment in attachments.items():
+                if isinstance(attachment, basestring):
+                    info = {"content": attachment}
+                elif "data" in attachment:
+                    # TODO find out if this is only needed for serialized test data
+                    info = {
+                        "content": base64.b64decode(attachment["data"]),
+                        "content_type": attachment.get("content_type"),
+                    }
+                else:
+                    # ignore attachment stub with no content
+                    continue
+                self.lazy_put_attachment(name=name, **info)
+        return self
+
+    def lazy_list_attachments(self):
+        keys = set()
+        keys.update(getattr(self, '_LAZY_ATTACHMENTS', None) or {})
+        keys.update(self.blobs or {})
+        return keys
+
+    def put_attachment(self, content, name=None, content_type=None,
+                       content_length=None):
+        self._LazyAttachmentDoc__remove_cached_attachment(name)
+        info = self._LazyAttachmentDoc__store_lazy_attachment(
+                        content, name, content_type, content_length)
+        return super(LazyBlobDoc, self).put_attachment(name=name, **info)
+
+    def save(self, **params):
+        if hasattr(self, '_PRE_SAVE'):
+            for pre_save in self._PRE_SAVE:
+                pre_save()
+
+            def del_pre_save():
+                del self._PRE_SAVE
+
+            self.register_post_save(del_pre_save)
+
+        if self._LAZY_ATTACHMENTS and self.__save_params is None:
+            assert self.__save_params is None, (self.__save_params, params)
+            self.__save_params = params
```

Comment:
nit: This doesn't seem to get used inside this `if` branch and gets deleted before exiting the branch. Is it used by `self.atomic_blobs`?